src/libtego_ui/ui/qml/MessageDelegate.qml

```diff
@@ -257,85 +247,22 @@ Column {
-        var object = hyperLinkLeftClickContextMenu.createObject(delegate, (link !== undefined) ? { 'hoveredLink' : link } : { })
-        // XXX FIXME QtQuickControls private API. The only other option is 'visible', and it is not reliable. See PR#183
-        object.popupVisibleChanged.connect(function() { if (!object.__popupVisible) object.destroy(1000) })
-        object.popup()
-    }
-
-    function showContextMenu(link) {
-        var object = rightClickContextMenu.createObject(delegate, (link !== undefined) ? { 'hoveredLink': link } : { })
-        // XXX FIXME QtQuickControls private API. The only other option is 'visible', and it is not reliable. See PR#183
-        object.popupVisibleChanged.connect(function() { if (!object.__popupVisible) object.destroy(1000) })
+    function showContextMenu() {
+        var object = rightClickContextMenu.createObject(delegate, { })
+        object.popupVisibleChanged.connect(function() { if (!object.visible) object.destroy(1000) })
         object.popup()
     }
 
-    Component {
-        id: hyperLinkLeftClickContextMenu
-
-        Menu {
-            property string hoveredLink: textField.hasOwnProperty('hoveredLink') ? textField.hoveredLink : ""
-            MenuItem {
-                //: Text for context menu command to copy a url to the clipboard
-                text: qsTr("Copy Link")
-                visible: hoveredLink.length > 0
-                onTriggered: LinkedText.copyToClipboard(hoveredLink)
-            }
-            MenuItem {
-                //: Text for context menu command to open a url in a web browser
-                text: qsTr("Open with Browser")
-                visible: hoveredLink.length > 0 && hoveredLink.substr(0,4).toLowerCase() == "http"
-                onTriggered: {
-                    if (uiSettings.data.alwaysOpenBrowser) {
-                        Qt.openUrlExternally(hoveredLink)
-                    } else {
-                        var window = uiMain.findParentWindow(delegate)
-                        var object = createDialog("OpenBrowserDialog.qml", { 'link': hoveredLink, 'contact': contact }, window)
-                        object.visible = true
-                    }
-                }
-            }
-        }
-    }
-
     Component {
         id: rightClickContextMenu
 
         Menu {
-            property string hoveredLink: textField.hasOwnProperty('hoveredLink') ? textField.hoveredLink : ""
-            MenuItem {
-                text: (hoveredLink.length > 0 && (hoveredLink.substr(0,9).toLowerCase() == "ricochet:")) ?
-                    //: Text for context menu command to copy a ricochet contact id to clipboard
-                    qsTr("Copy ID") :
-                    //: Text for context menu command to copy a url to the clipboard
-                    qsTr("Copy Link")
-                visible: hoveredLink.length > 0
-                onTriggered: LinkedText.copyToClipboard(hoveredLink)
-            }
```

Comment:
Should this menu item be retained? i.e. you have one menu item that's "Copy Link" and the other is "Copy Message"

Reply:
This patch completely removes generating hyperlinks from URIs. The remaining menu items should be 'Copy Selection' and 'Copy Message'f

Reply:
Ahhh my bad